public/app/plugins/datasource/loki/querybuilder/components/LabelBrowserModal.tsx

```diff
@@ -36,8 +37,15 @@ export const LabelBrowserModal = (props: Props) => {
     onClose();
   };
 
+  const reportClose = (method: string) => {
+    reportInteraction('grafana_loki_label_browser_closed', {
+      app: app,
+      method: method,
+    });
+  };
+
   return (
-    <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>
+    <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose} reportClose={reportClose}>
```

Comment:
Hm, isn't `onDismiss` the callback that can be used here? I know that we don't get the way how the modal was closed by that, but it would also be not as "intrusive" into the Modal's code.

What do you think about adding the feature tracking in LabelBrowserModal to `onDismiss` and ask User Essentials about adding the close-type to that callback?